examples/platform/silabs/BaseApplication.cpp

```diff
@@ -935,8 +941,10 @@ void BaseApplication::OnPlatformEvent(const ChipDeviceEvent * event, intptr_t)
         // Update the LCD screen with SSID and connected state
         if (screen == SilabsLCD::Screen_e::StatusScreen)
         {
-            BaseApplication::UpdateLCDStatusScreen(false);
-            AppTask::GetLCD().SetScreen(screen);
+            AppEvent event;
+            event.Type    = AppEvent::kEventType_LCD;
+            event.Handler = AppTask::GetAppTask().UpdateDisplayHandler;
+            PostEvent(&event);
```

Comment:
Should we do a `PostUpdateDisplayEvent() `function if we repeat it a few times

Reply:
The point if an function is introduced the event data can not be futhur changed so it would be better in my opinion to post it as it is, since this way we can pass the required event data if needed.

Reply:
I am not sure if I understood exactly what you meant, but from my code review, we would just have to pass as a argument to this postLcdUpdateEvent the Screen_e enum value

Reply:
Added the PostUpdateDisplayEvent